oauth2ns.go

```diff
@@ -143,8 +144,9 @@ func startHTTPServer(ctx context.Context, conf *oauth2.Config, appConfig Config)
 	stopHTTPServerChan = make(chan struct{})
 	cancelAuthentication = make(chan struct{})
 
-	http.HandleFunc(fmt.Sprintf("%s",appConfig.Path), callbackHandler(ctx, conf, clientChan,appConfig))
-	srv := &http.Server{Addr: ":" + strconv.Itoa(appConfig.Port)}
+  router := mux.NewRouter()
```

Comment:
is there a reason to choose gorilla over like chi (seems like 6 of one, half a dozen of another)?

Would it be possible to provide some evidence of the before/after this to show its working with this changeset, or at least describe the specific scenario you ran into, and confirm that you did re-try with this change and it fixed the problem. 

thanks!

Reply:
Heya, 

It doesn't really matter what handler / router was used because we're never really engaging it. I just wanted to pick something commonly used by the community *if* someone wanted to expand upon it. The inherent problem was that originally:

* A new http.Server object is created
* the *global* http handler is then used for the callback func. 

In the scope of GOAS this causes a problem whenever re-authorization would happen since it would generate a panic. We're just tossing gorilla in as the handler for the server and setting the handle func directly on the server. 

![goas (3)](https://user-images.githubusercontent.com/952951/130386283-6985a3aa-3a28-4b35-ab67-e04d76a2e0a8.gif)

The above GIF shows that we can re-authorize multiple times without incurring a panic and killing GOAS. Also highlights the new menu items for accessing the Access Token, Refresh Token and manual reauthorization

Reply:
awesome, thanks for the extra detail!